ingestion/src/metadata/orm_profiler/api/workflow.py

```diff
@@ -230,22 +277,34 @@ def copy_service_config(self, database) -> None:
             if hasattr(self.config.source.serviceConnection.__root__.config, "catalog"):
                 copy_service_connection_config.catalog = database.name.__root__
 
-        self.create_processor(copy_service_connection_config)
+        return copy_service_connection_config
 
     def execute(self):
         """
         Run the profiling and tests
         """
 
-        for database in self.get_database_entities():
-            try:
-                self.copy_service_config(database)
+        databases = self.get_database_entities()
+
+        if not databases:
+            raise ValueError(
+                "databaseFilterPattern returned 0 result. At least 1 database must be returned by the filter pattern."
+                f"\n\t- includes: {self.source_config.databaseFilterPattern.includes}\n\t- excludes: {self.source_config.databaseFilterPattern.excludes}"
+            )
 
+        for database in databases:
+            copied_service_config = self.copy_service_config(database)
+            try:
                 for entity in self.get_table_entities(database=database):
                     try:
-                        profile_and_tests: ProfilerResponse = self.processor.process(
+                        self.create_profiler_interface(copied_service_config, entity)
+                        self.create_profiler_obj()
+                        profile = self.profiler_obj.process()
```

Comment:
Ah ok took me a while to figure out this logic here. Instead of creating the interface and profiler and store is a class attribute, could we maybe pass the information as arguments?

For example in the `print_status` below, you would be using just the latest processor_obj instead of keeping the status of each `database` if I am following this correctly. Maybe we can clean this up eventually (does not need to happen now)

Reply:
@pmbrull yes, that makes sense. I'll update this in the workflow PR. As far as the `processor_obj` will be deleted and we'll have a `test_something` object where we'll implement new logic.